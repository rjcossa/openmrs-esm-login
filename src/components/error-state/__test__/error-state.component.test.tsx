import * as React from 'react';
import { cleanup, screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import EmptyState from '../error-state.component';

afterEach(cleanup);

const error = {
  response: {
    status: 200,
    statusText: 'There was an issue getting the information from the server',
  },
};

it('Test if component is able to render the title', () => {
  render(<EmptyState headerTitle="Header" error={error} />);
  const headerElement = screen.getByTestId('header-title');
  expect(headerElement).toBeInTheDocument();
});

it('Test if component matches snapshot', () => {
  const tree = renderer.create(<EmptyState headerTitle="Header" error={error} />).toJSON();
  expect(tree).toMatchSnapshot();
});
