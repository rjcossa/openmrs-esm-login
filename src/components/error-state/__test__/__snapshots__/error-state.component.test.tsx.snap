// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if component matches snapshot 1`] = `
<div
  className="bx--tile bx--tile--light tile"
>
  <h1
    className="heading"
    data-testid="header-title"
  >
    Header
  </h1>
  <p
    className="errorMessage"
  >
    Error
     
    200: 
    There was an issue getting the information from the server
  </p>
  <p
    className="errorCopy"
  >
    Sorry, there was a problem displaying this information. You can try to reload this page, or contact the site administrator and quote the error code above.
  </p>
</div>
`;
